enhancements/ingress/set_eip_nlb_ingress.md

```diff
@@ -0,0 +1,983 @@
+5. The Ingress Operator will resolve the `LoadBalancerProgressing` condition back to
+   `Status: False` as long as `EIPAllocations` is equivalent to `service.beta.kubernetes.io/aws-load-balancer-eip-allocations`.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+Currently, there is no unique consideration for making this change work with Hypershift.
+
+#### Standalone Clusters
+
+This proposal does not have any special implications for standalone clusters.
+
+#### Single-node Deployments or MicroShift
+
+This proposal does not have any special implications for single-node
+deployments or MicroShift.
+
+### Implementation Details/Notes/Constraints
+
+1. #### Set EIP through installer for the default ingress controller:
+   1. The admin sets the `eipAllocations` in the installer using the install-config.yaml.
+   2. The installer then sets the ingress cluster object with the `eipAllocations`. 
+   3. The cluster ingress operator then creates a default ingress controller by:
+      1. creating an default ingress controller CR.
+      2. then scaling a deployment for the Ingress Controller CR which has the value of the `eipAllocations` from the 
+     ingress cluster object.
+      3. then creating a service of service type load balancer with the annotation `service.beta.kubernetes.io/aws-load-balancer-eip-allocations`, 
+     which uses the value from the field `eipAllocations` of ingress controller CR.
+
+#### 2. Create a new ingress controller with EIPs:
+   
+   The admin sets the `eipAllocations` in the Ingress Controller CR. Then same process
+   as per point 1.iii.b and 1.iii.c is followed by the ingress operator. 
+
+#### 3. Add, delete, or change the EIP for an existing IngressController:
+   
+   When the admin changes the eipAllocations field in the Ingress Controller CR, the ingress operator sets the Load Balancer
+   Progressing status as per [section](#Cluster administrator updates eipAllocations field of the existing ingress controller)
+   The cluster admin has to manually delete service in order to effectuate the change.
+   Then step 1.iii.c is followed by the ingress operator.
+
+   If the admin wants to detect the change in `eipAllocations` and update the LB service automatically,
+   rather than explicitly deleting the Service after changes, they can add the  `ingress.operator.openshift.io/auto-delete-load-balancer annotation`.
+   
+### 4. Any change to the service annotation does not trigger the operator to manage the load balancer service.
+  The `status.endpointPublishingStrategy.loadBalancer.providerParameters.aws.networkLoadBalancer.eipAllocations`
+  will eventually reflect the configured subnet value by mirroring the value of
+  `service.beta.kubernetes.io/aws-load-balancer-eip-allocations` on the service. There are
+  two scenarios in which the `Subnets` status won't be equal to the `Subnets` spec:
```

Comment:
I got confused for a second here thinking I was review my EP. I think you forgot to change all of the references from `Subnets` to `eipAllocations` :smiley:

Reply:
sorry

Reply:
done